tests/core/scenario/test_scenario_manager.py

```diff
@@ -317,7 +317,7 @@ def test_notification_subscribe(mocker):
     mocker.patch.object(_utils, "_load_fct", side_effect=[notify_1, notify_2])
 
     # test subscribing notification
-    _ScenarioManager._subscribe(notify_1, scenario)
+    _ScenarioManager._subscribe(callback=notify_1, scenario=scenario)
```

Comment:
Can you create a test where the callback takes multiple params?

Reply:
Yes! I will add the tests in the the second half of this PR which handles the jobs.